site/utils/formUtils.ts

```diff
@@ -66,3 +68,39 @@ export const sortStops = (stops: Stop[]) => {
         }
     });
 };
+
+export const getEndingOnDateText = (
+    disruptionRepeats: string | undefined,
+    disruptionRepeatsEndDate: string | undefined,
+    disruptionStartDate?: string,
+    disruptionEndDate?: string,
+) => {
+    if (disruptionRepeats === "weekly" && disruptionStartDate && disruptionRepeatsEndDate && disruptionEndDate) {
+        const startDateDay = getFormattedDate(disruptionStartDate).day();
+        const endDateDay = getFormattedDate(disruptionEndDate).day();
+        const endingOnDateDay = getFormattedDate(disruptionRepeatsEndDate).day();
+
+        const diffInDaysWithEndingOn =
+            startDateDay < endingOnDateDay ? endingOnDateDay - startDateDay : endingOnDateDay - startDateDay + 7;
```

Comment:
As this is dealing with dates, would it be better suited to be in the dates file?